### DISCUSSION TITLE
I created a NodeJS wrapper around ODD! 🥳

### DISCUSSION THREAD

**Discussion by Chaphasilor** [KoalaBear84/OpenDirectoryDownloader#63] [Show and tell]
Because I'm using ODD in an automated way, but ODD is a CLI tool, I created a wrapper for JavaScript/NodeJS so it's easier to use through code!

**It's available on npm: https://www.npmjs.com/package/open-directory-downloader**
I hope there are some fellow NodeJS developers out there who will find this useful :)

Right now the only thing supported is scanning an OD and getting the results, but that works quite well.  
I plan to add some more features as requested/needed by myself, like events (progress, end, "milestones" like start of speedtest, etc.) or more options.

Let me know what you think! 😁

**Comment by KoalaBear84** [comment]
Haha, great, I've seen it come online on your GitHub 1-2 days ago 😂

I don't have NodeJS, I did use it previously when you got an issue with it. 😇

Maybe someone with experience with it can test it. O lr is it easy to do? Or maybe with a Docker I guess?

**Reply by Chaphasilor** [reply]
Haha yeah it was my project for the weekend 😅

If you have NodeJS installed (and therefore `npm`), you can try it out pretty easily:

1. Create a new folder, `cd` into it and run `npm init -y`
2. Run `npm i open-directory-downloader`
3. Once installed (it should also download ODD v1.9.3.3, please let me know if there are issues), create a new file called `test.js`
4. Put the following into `test.js`:
  ```js
  let OpenDirectoryDownloader = require(`open-directory-downloader`)
  let ODD = new OpenDirectoryDownloader()
  ODD.scanUrl(`https://www.culture-informatique.net/download/`).then(console.log).catch(console.error)
  ```
5. Run `node test.js`

If all goes well you should see some JSON output at the end :D
